CDEFramework/Libraries/LegacyDictionarySupport/GlossaryAPIClient.cs

```diff
@@ -0,0 +1,152 @@
+
+            HttpClient client = new HttpClient();
+            //NOTE: the base URL MUST have a trailing slash
+            client.BaseAddress = new Uri(String.Format("{0}/{1}/{2}/", baseApiPath, appPath, versionPath));
+
+            this._client = client;
+        }
+
+        /// <summary>
+        /// Calls the GetById endpoint of the Glossary API
+        /// </summary>
+        /// <param name="dictionary">Dictionary name (required)</param>
+        /// <param name="audience">Audience type(required)</param>
+        /// <param name="language">Language to use (required)</param>
+        /// <param name="id">The ID of the term</param>
+        /// <param name="useFallback">Whether or not to use the fallback logic</param>
+        /// <returns>A Best Bet result</returns>
+        public GlossaryTerm GetById(
+            string dictionary,
+            string audience,
+            string language,
+            string id,
+            bool useFallback = false 
+            )
+        {
+            //GlossaryTerm resultTerm = null;
+
+            // Check fields
+            if (String.IsNullOrWhiteSpace(dictionary))
+            {
+                throw new ArgumentNullException("The dictionary is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(audience))
+            {
+                throw new ArgumentNullException("The audience is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(language))
+            {
+                throw new ArgumentNullException("The language is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentNullException("The ID is null or an empty string");
+            }
+
+            // Set up search param string: {dictionary}/{audience}/{language}/{id:long}
+            string[] searchParams = { dictionary, audience, language, id, useFallback.ToString() };
+            string searchParam = string.Format("{0}/{1}/{2}/{3}?useFallback={4}", searchParams);
+
+            //Get the HTTP response content from GET request
+            HttpContent httpContent = ReturnGetRespContent("Terms", searchParam);
+            if (httpContent != null)
+            {
+                Task<GlossaryTerm> term = ReadAsJsonAsync<GlossaryTerm>(httpContent);
+                var resultTerm = term.Result;
+                return resultTerm;
+            }
+            else
+            {
+                return null;
+            }
+            
+        }
+
+        /// <summary>
+        /// Gets the response content of a GET request.
+        /// </summary>
+        /// <param name="path">Path for client address</param>
+        /// <param name="param">Param in URL</param>
+        /// <returns>HTTP response content</returns>
+        public HttpContent ReturnGetRespContent(String path, String param)
```

Comment:
Should include a mention of a `null` return in case the HTTP call doesn't succeed.